File: (04-02-2024)Valid Anagram/joshlin/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        frequency = {}
        for c in s:
            frequency[c] = frequency.get(c, 0) + 1
        for c in t:
            if c not in frequency:
                return False
            frequency[c] -= 1
        for count in frequency.values():
            if count != 0:
                return False
        return True

File: (04-02-2024)Valid Anagram/joshlin/solution.ts
function isAnagram(s: string, t: string): boolean {
  const frequency = new Map<string, number>();
  for (const char of s) {
      frequency.set(char, (frequency.get(char) ?? 0) + 1);
  };
  for (const char of t) {
      if (!frequency.has(char)) return false;
      frequency.set(char, frequency.get(char) - 1);
  };
  for (const count of frequency.values()) {
      if (count != 0) return false;
  };
  return true;
};